File: src/main/java/com/sothawo/springdataelastictest/MovieRepository.java
/*
 * (c) Copyright 2019 sothawo
 */
package com.sothawo.springdataelastictest;

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;

import java.util.List;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
public interface MovieRepository extends ElasticsearchRepository<Movie, String> {

    List<Movie> findByTitle(String title);

    List<Movie> findFirst3ByTitle(String title);

    List<Movie> findFirst3ByTitleOrderByYearDesc(String title);
}

File: src/main/kotlin/com/sothawo/springdataelastictest/MovieRepository.kt
/*
 * (c) Copyright 2019 sothawo
 */
package com.sothawo.springdataelastictest

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
interface MovieRepository : ElasticsearchRepository<Movie, String> {

    fun findByTitle(title: String): List<Movie>

    fun findFirst3ByTitle(title: String): List<Movie>

    fun findFirst3ByTitleOrderByYearDesc(title: String): List<Movie>
}
